ginivision/src/doc/source/guide-for-open-with.rst

```diff
@@ -0,0 +1,182 @@
+        ...
+    }
+ 
+The Intent will contain an ``Uri`` pointing to the PDF or image. The ``Uri`` can be in the ``data`` field or since Android 4.1 (API Level 16) it can also be in the ``clipData`` field. 
+
+.. note:: 
+
+    To also support `ACTION_VIEW <https://developer.android.com/reference/android/content/Intent.html#ACTION_VIEW>`_ we recommend using ``data`` and ``clipData``. The `Androd example <https://developer.android.com/training/sharing/receive.html>`_ uses ``intent.getParcelableExtra(Intent.EXTRA_STREAM)`` to extract the ``Uri``, but this works only with `ACTION_SEND <https://developer.android.com/reference/android/content/Intent.html#ACTION_SEND>`_. 
+
+Getting the ``Uri`` from the Intent:
+
+.. code-block:: java
+
+    Uri uri = intent.getData();
+    if (uri == null && Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
+        ClipData clipData = intent.getClipData();
+        if (clipData != null && clipData.getItemCount() > 0) {
+            uri = clipData.getItemAt(0).getUri();
+        }
+    }
+
+We recommend checking the MIME type of Intent. The MIME type can be in the ``type`` field or since Android 4.1 (API Level 16) it can also be in the ``clipData`` field.
+ 
+Getting the MIME type from the Intent:
+
+.. code-block:: java
+
+    List<String> mimeTypes = new ArrayList<>();
+    String type = intent.getType();
+    if (mimeType != null) {
+        mimeTypes.add(type);
+    } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
+        ClipData clipData = intent.getClipData();
+        if (clipData != null) {
+            ClipDescription description = clipData.getDescription();
+            for (int i = 0; i < description.getMimeTypeCount(); i++) {
+                type = description.getMimeType(i);
+                mimeTypes.add(type);
+            }
+        }
+    }
+
+Having the ``Uri`` the contents of the PDF or image files can be read using the ``ContentResolver``.
+ 
+Reading the PDF or image into a byte array:
+
+.. code-block:: java
+
+    byte[] bytes = null;
+    ContentResolver contentResolver = activity.getContentResolver()
+    InputStream inputStream = null;
+    try {
+        inputStream = contentResolver.openInputStream(uri);
+        if (inputStream != null) {
+            // ByteStreams is a utility class from Google’s Guava library
+            bytes = ByteStreams.toByteArray(inputStream);
+        }
+    } finally {
+        if (inputStream != null) {
+            inputStream.close();
+        }
+    }
+
+The byte array from the example above can be directly uploaded to the Gini API for information extraction.
+
+Documentation
+^^^^^^^^^^^^^
+
+- Android: https://developer.android.com/training/sharing/receive.html
+- Gini API: http://developer.gini.net/gini-api/html/documents.html#submitting-files
+- Gini API SDK: http://developer.gini.net/gini-sdk-android/guides/common-tasks.html#upload-a-document
+
+Showing a preview of the PDF’s first page
```

Comment:
Please see comment from iOS documentation version. Since we don't currently support multipage scan, do we want to only speak of a preview for the PDF/image, instead of the first page of the PDF? And also making this consistent in the following text.